### PR TITLE
Add C api for methods that are required to handle connection migratio…

### PR DESCRIPTION
**Pull request by normanmaurer**
…n on the server side.

Motivation:

We need to expose some methods via the C api to be able to over the remote peer connection ids that it can be used for migration

Modifications:

Add missing ffi

Result:

Be able to handle connection migrations

### CODE REVIEW

**Comment by normanmaurer** [comment]
Depending on if https://github.com/cloudflare/quiche/pull/1655 is accepted and merged we might want to rename some c functions here

File: quiche/include/quiche.h [hunk 1]
```diff
@@ -678,6 +678,30 @@ ssize_t quiche_conn_send_ack_eliciting_on_path(quiche_conn *conn,
                            const struct sockaddr *local, size_t local_len,
                            const struct sockaddr *peer, size_t peer_len);
 
+// Returns true if there are retired source connection ids and fill the the parameters
```
Comment (FranzBusch) [inline quiche/include/quiche.h:681]:
```suggestion
// Returns true if there are retired source connection ids and fill the parameters
```

**Comment by normanmaurer** [comment]
Seems like the test failure is not related... @LPardue @ghedo thoughts ?

**Comment by ghedo** [comment]
The clippy error should be fixed by https://github.com/cloudflare/quiche/pull/1659

**Comment by normanmaurer** [comment]
rebased and pushed

**Review by LPardue — DISMISSED** [review]

**Comment by normanmaurer** [comment]
@ghedo rebased

**Review by ghedo — APPROVED** [review]

**Pull request merged**